gtest/src/error.rs

```diff
@@ -0,0 +1,97 @@
+// This file is part of Gear.
+
+// Copyright (C) 2022 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use anyhow::Error as AnyhowError;
+use codec::Error as CodecError;
+use core_processor::ProcessorError;
+use gear_core::{ids::ProgramId, memory::WasmPageNumber};
+use wasmtime::MemoryAccessError;
+
+pub type TestResult<T> = Result<T, TestError>;
```

Comment:
I think `Test` prefix is unnecessary as for error

Reply:
This was made to disambiguate it with `core::result::Result` (that is used in `pub trait WasmProgram` for example), `anyhow::Result`, etc. I can rename `TestResult` to `Result` but this forces `gtest` users to disambiguate them manually. I don't think it is a good idea.

Reply:
> to disambiguate it with core::result::Result

We can define it as 
```
pub type Result<T, E = Error> = std::result::Result<T, E>
```

> `anyhow::Result`, etc

I see only `anyhow::Error` is used as variant of `TestError`


Thus, I guess there will be no ambiguity

Reply:
Ah, agree, generic alias with default `E` allows using both `Result<T>` and `Result<T, E>`. Fixed.